addons/web/controllers/main.py

```diff
@@ -990,140 +990,155 @@ def action(self, model, id):
             'action', 'tree_but_open',[(model, id)],
             False)
 
-class Binary(http.Controller):
+def binary_content(xmlid=None, model='ir.attachment', id=None, field='datas', unique=False,
+        filename=None, filename_field='datas_fname', download=False, mimetype=None, default_mimetype='application/octet-stream', env=None):
+    """ Get file, attachment or downloadable content
+
+    If the ``xmlid`` and ``id`` parameter is omitted, fetches the default value for the
+    binary field (via ``default_get``), otherwise fetches the field for
+    that precise record.
+
+    :param str xmlid: xmlid of the record
+    :param str model: name of the model to fetch the binary from
+    :param int id: id of the record from which to fetch the binary
+    :param str field: binary field
+    :param bool unique: add a max-age for the cache control
+    :param str filename: choose a filename
+    :param str filename_field: if not create an filename with model-id-field
+    :param bool download: apply headers to download the file
+    :param str mimetype: mintype of the field (for headers)
+    :param str default_mimetype: default mintype if no mintype found
+    :param Environment env: by default use request.env
+    :returns: (status, headers, content)
+    """
+    status = 404
+    # get object and content
+    obj = None
+    content = None
+    try:
+        if xmlid:
+            obj = request.env.ref(xmlid, False)
+            if hasattr(obj, field):
+                status = 200
+        elif id:
+            obj = request.env[model].browse(int(id))
+            if hasattr(obj, field):
+                status = 200
+        else:
+            content = request.env[model].default_get([field]).get(field, None)
+            if content:
+                status = 200
+    except:
```

Comment:
... all made mistakes ... it could be xmlid unknown, unknown model, an access right, field that is not good, id that is not an id